onboarding_project/squad_pantry_app/models.py

```diff
@@ -150,3 +149,62 @@ class ConfigurationSettings(models.Model):
 
     def __str__(self):
         return self.constant
+
+
+class PerformanceMetrics(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True, unique=True)
+    average_throughput = models.IntegerField(editable=False)
+    average_turnaround_time = models.CharField(max_length=256, editable=False)
+
+    @classmethod
+    def calculate_avg_performance_metrics(cls):
+        if PerformanceMetrics.objects.all().count() < 1:
+            completed_orders_curr_day = Order.objects.filter(
+                status=Order.DELIVERED,
+                closed_at__isnull=False
+            )
+        else:
+            start_date_time = PerformanceMetrics.objects.latest('id').created_at
+            completed_orders_curr_day = Order.objects.filter(
+                status=Order.DELIVERED,
+                closed_at__isnull=False,
+                created_at__range=(start_date_time, timezone.now())
+            )
+        throughput = len(completed_orders_curr_day)
+
+        if throughput == 0:
+            return 0, '00:00:00.00'
+
+        total_turnaround_time = 0
+        for obj in completed_orders_curr_day:
+            total_turnaround_time += (obj.closed_at - obj.created_at).total_seconds()
+
+        average_turnaround_time = total_turnaround_time/throughput
+        return timedelta(seconds=average_turnaround_time), throughput
+
+    @classmethod
+    def create_avg_performance_metrics(cls):
+        turnaround_time, throughput = PerformanceMetrics.calculate_avg_performance_metrics()
+        print(turnaround_time, throughput)
+        PerformanceMetrics.objects.create(average_throughput=throughput, average_turnaround_time=turnaround_time)
+
+    @classmethod
+    def get_metrics_data(cls, start_date, end_date):
+        metrics = PerformanceMetrics.objects.filter(created_at__range=(start_date, end_date))
```

Comment:
The main way we query `PerformanceMetrics` table is by `created_at`. If we change our metrics calculation interval to 1 hour instead of 1 day, we can potentially end up with thousands of records in the metrics table. In that case the above query would run really slow. 

- Why would the above query be slow if the performance metrics table had say, a million rows (Metrics for an entire year maybe?)
- How can we make the above query faster?